tools/txg/TXG/Types.hs

```diff
@@ -83,118 +75,134 @@ data TimingDistribution
-    <*< nodeChainIds        ..: "nodeChainIds"        % o
-    <*< isChainweb          ..: "isChainweb"          % o
-    <*< hostAddresses       ..: "hostAddresses"       % o
-    <*< nodeVersion         ..: "chainwebVersion"     % o
-    <*< logHandleConfig     ..: "logging"             % o
-
-defaultScriptConfig :: ScriptConfig
-defaultScriptConfig = ScriptConfig
-  { _scriptCommand   = RunSimpleExpressions def def
+  , _logHandleConfig :: !U.HandleConfig
+  , _batchSize :: !BatchSize
+  } deriving (Show, Generic)
+
+scriptCommand :: Functor f => (TXCmd -> f TXCmd) -> Args -> f Args
+scriptCommand f sc = (\tc -> sc { _scriptCommand = tc }) <$> f (_scriptCommand sc)
+
+nodeChainIds :: Functor f => ([ChainId] -> f [ChainId]) -> Args -> f Args
+nodeChainIds f sc = (\tc -> sc { _nodeChainIds = tc }) <$> f (_nodeChainIds sc)
+
+isChainweb :: Functor f => (Bool -> f Bool) -> Args -> f Args
+isChainweb f sc = (\tc -> sc { _isChainweb = tc }) <$> f (_isChainweb sc)
+
+hostAddresses :: Functor f => ([HostAddress] -> f [HostAddress]) -> Args -> f Args
+hostAddresses f sc = (\tc -> sc { _hostAddresses = tc }) <$> f (_hostAddresses sc)
+
+nodeVersion :: Functor f => (ChainwebVersion -> f ChainwebVersion) -> Args -> f Args
+nodeVersion f sc = (\tc -> sc { _nodeVersion = tc }) <$> f (_nodeVersion sc)
+
+logHandleConfig :: Functor f => (U.HandleConfig -> f U.HandleConfig) -> Args -> f Args
+logHandleConfig f sc = (\tc -> sc { _logHandleConfig = tc }) <$> f (_logHandleConfig sc)
+
+batchSize :: Functor f => (BatchSize -> f BatchSize) -> Args -> f Args
+batchSize f sc = (\tc -> sc { _batchSize = tc }) <$> f (_batchSize sc)
+
+instance ToJSON Args where
+  toJSON o = object
+    [ "scriptCommand"   .= _scriptCommand o
+    , "nodeChainIds"    .= _nodeChainIds o
+    , "isChainweb"      .= _isChainweb o
+    , "hostAddresses"   .= _hostAddresses o
+    , "chainwebVersion" .= _nodeVersion o
+    , "logHandle"       .= _logHandleConfig o
+    , "batchSize"       .= _batchSize o
+    ]
+
+instance FromJSON (Args -> Args) where
+  parseJSON = withObject "Args" $ \o -> id
+    <$< scriptCommand   ..: "scriptCommand"   % o
+    <*< nodeChainIds    ..: "nodeChainIds"    % o
+    <*< isChainweb      ..: "isChainweb"      % o
+    <*< hostAddresses   ..: "hostAddresses"   % o
+    <*< nodeVersion     ..: "chainwebVersion" % o
+    <*< logHandleConfig ..: "logging"         % o
+    <*< batchSize       ..: "batchSize"       % o
+
+defaultArgs :: Args
+defaultArgs = Args
+  { _scriptCommand   = RunSimpleExpressions def
   , _nodeChainIds    = []
   , _isChainweb      = True
   , _hostAddresses   = [unsafeHostAddressFromText "127.0.0.1:1789"]
   , _nodeVersion     = v
-  , _logHandleConfig = U.StdOut }
+  , _logHandleConfig = U.StdOut
+  , _batchSize       = BatchSize 1 }
   where
     v :: ChainwebVersion
     v = fromJuste $ chainwebVersionFromText "timedCPM-peterson"
 
-scriptConfigParser :: MParser ScriptConfig
+scriptConfigParser :: MParser Args
```

Comment:
In the context of `pLeftSemigroupalUpdate` it may be more convenient to define something along the lines of

```haskell
<*< hostAddresses %:: pListSemigroupalUpdate % textOption
    % long "host"
    <> short 'h'
    <> metavar "HOSTNAME:PORT"
    <> help "hostname and port of a host the receives transactions. This option can be used multiple times to send  transactions to more than one host"
```

i.e. bundle the hostname and port in a single option, which might be more intuitive for options that can be provided multiple times.

Reply:
That's currently already the case, is it not?